### PR TITLE
feat: draft of google authenticator enroll [WIP]

### PR DESCRIPTION
**Pull request by TianGan-okta**
## PR Checklist

- [ ] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Issue:

- [OKTA-XXXXXX](https://oktainc.atlassian.net/browse/OKTA-XXXXXX)

### CODE REVIEW

File: playground/mocks/config/responseConfig.js [hunk 1]
```diff
@@ -4,19 +4,19 @@ const idx = {
   // ===== IDX
 
   '/idp/idx/introspect': [
-    'identify',
+    // 'identify',
```
Comment (pradeepdewda-okta) [inline playground/mocks/config/responseConfig.js:7]: you don't need to push these change in playground.. keep new entries commented!

File: packages/@okta/i18n/src/properties/login.properties [hunk 2]
```diff
@@ -799,6 +799,14 @@ oie.okta_verify.push.sent = Push notification sent
 oie.okta_verify.push.send = Send push notification
 oie.okta_verify.signed_nonce.label = Use Okta Verify on this device
 
+## Google authenticator
+oie.enroll.google_authenticator.setup.title = Set up Google Authenticator
+oie.enroll.google_authenticator.scanBarcode.title = Scan Barcode
```
Comment (pradeepdewda-okta) [inline packages/@okta/i18n/src/properties/login.properties:804]: should be `Scan barcode` ?

File: packages/@okta/i18n/src/properties/login.properties [hunk 3]
```diff
@@ -799,6 +799,14 @@ oie.okta_verify.push.sent = Push notification sent
...
+oie.enroll.google_authenticator.scanBarcode.description = Launch Google Authenticator, tab "+" icon, than select "Scan barcode".
+oie.enroll.google_authenticator.scanBarcode.cannotScan = Can't scan?
+oie.enroll.google_authenticator.scanBarcode.cannotScanBarcode = Can't scan barcode?
+oie.enroll.google_authenticator.manualSetupInstructions = To set up manually enter your Account username and then input the following in the Secret Key Field
```
Comment (pradeepdewda-okta) [inline packages/@okta/i18n/src/properties/login.properties:808]: `Account` -> `account` 
`Secret Key Field` -> `secret key field` 

Confirm with UX if they need them capitalized

File: src/v2/view-builder/views/google-authenticator/EnrollGoogleAuthenticatorBarcodeView.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        <div class="qrcode-container">
+          <img class="qrcode" src={{href}} alt="qr code"></img>
+          <a href="#" class="switch-channel-link">
+            {{i18n code="oie.enroll.google_authenticator.scanBarcode.cannotScan" bundle="login"}}
+          </a>
+        </div>
+        </div>
+      {{/if}}
+    `,
+  initialize () {
```
Comment (pradeepdewda-okta) [inline src/v2/view-builder/views/google-authenticator/EnrollGoogleAuthenticatorBarcodeView.js:22]: add a new line after function definition.

File: src/v2/view-builder/views/google-authenticator/EnrollGoogleAuthenticatorBarcodeView.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    View.prototype.initialize.apply(this, arguments);
+    this.options.appState.set('isScanBarcode', true);
+  },
+  getTemplateData () {
+    const contextualData = this.options.appState.get('currentAuthenticator').contextualData;
+    const isScanBarcode = this.options.appState.get('isScanBarcode');
+    return { href: isScanBarcode && contextualData.qrcode?.href };
+  },
+  postRender () {
+    this.$el.find('.switch-channel-link').on('click', (event) => {
```
Comment (pradeepdewda-okta) [inline src/v2/view-builder/views/google-authenticator/EnrollGoogleAuthenticatorBarcodeView.js:32]: Don't recommend to put an event handler through DOM.. I suggest to find some alternative way to do it.

File: src/v2/view-builder/views/google-authenticator/EnrollGoogleAuthenticatorController.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import EnrollAuthenticatorMannuallySetupView from './EnrollAuthenticatorMannuallySetupView';
+import FormType from '../../../../util/FormType';
+
+
+export default BaseForm.extend({
+  hasSavingState: false,
+  title () {
+    return null;
+  },
+  initialize () {
```
Comment (pradeepdewda-okta) [inline src/v2/view-builder/views/google-authenticator/EnrollGoogleAuthenticatorController.js:15]: add a new line

File: src/v2/view-builder/views/google-authenticator/EnrollGoogleAuthenticatorController.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    this.$('.shared-secret').show();
+    this.$('.oie-enroll-google-authenticator-manual-setup').show();
+  },
+
+  getUISchema () {
+    return [];
+  },
+
+  saveForm () {
+    console.log('asdasdasd');
```
Comment (pradeepdewda-okta) [inline src/v2/view-builder/views/google-authenticator/EnrollGoogleAuthenticatorController.js:50]: delete it :)

File: src/v2/view-builder/views/google-authenticator/EnrollGoogleAuthenticatorController.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+  },
+  className: 'oie-enroll-google-authenticator',
+
+  modelEvents: {
+    'triggerBarcode': '_triggerBarcode'
+  },
+
+  render () {
+    BaseForm.prototype.render.apply(this, arguments);
+    this.$('.oie-enroll-google-authenticator-manual-setup').hide();
```
Comment (pradeepdewda-okta) [inline src/v2/view-builder/views/google-authenticator/EnrollGoogleAuthenticatorController.js:35]: want to understand why we need to do this and find a better way to show/hide them.. Is it a toggle ?

File: src/v2/view-builder/views/google-authenticator/EnrollGoogleAuthenticatorController.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    this.$('.shared-secret').hide();
+  },
+
+  _triggerBarcode () {
+    this.$('.oie-enroll-google-authenticator-barcode').hide();
```
Comment (pradeepdewda-okta) [inline src/v2/view-builder/views/google-authenticator/EnrollGoogleAuthenticatorController.js:40]: same here..

**Comment by TianGan-okta** [comment]
Thanks @pradeepdewda-okta for a quick review. This is a WIP pr for internal using. Closing it..

**Pull request closed without merging**